### PR TITLE
[Code Change] AzCopy Error Improvements

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Added Error Code URL as seen below, proposed by Daniel
`azcopy copy "sourceurl" "destinationurl"`
```
failed to perform copy command due to error: cannot start job due to error: cannot list files due to reason GET https://siminmaintest.blob.core.windows.net/test
--------------------------------------------------------------------------------
RESPONSE 401: 401 Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
ERROR CODE: NoAuthenticationInformation
--------------------------------------------------------------------------------
﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>NoAuthenticationInformation</Code><Message>Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
RequestId:36f56f01-101e-000f-124c-8c69c6000000
Time:2024-04-11T20:13:02.1634782Z</Message></Error>
--------------------------------------------------------------------------------
.
ERROR DETAILS: https://aka.ms/AzCopyError/NoAuthenticationInformation
```

Added error code URLs to commonly used commands: copy, sync, remove, and list.

- E2E Tests added will only test NoAuthenticationInformation (and ResourceNotFound). Other error codes require extremely specific setups

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
@siminsavani-msft linked are active

**Review by pranavmalik-msft — COMMENTED** [review]
You may want to update your branch since I've included the fix for the failing test case.

**Review by pranavmalik-msft — CHANGES_REQUESTED** [review]
Can we have E2E testcases to verify this behavior

File: e2etest/newe2e_task_resourcemanagement.go [hunk 1]
```diff
@@ -220,3 +220,26 @@ func ValidateErrorOutput(a Asserter, stdout AzCopyStdout, errorMsg string) {
 	fmt.Println(stdout.String())
 	a.Error("expected error message not found in azcopy output")
 }
+
+func ValidateMultipleErrors(a Asserter, stdout AzCopyStdout, errorMsg []string) {
```
Comment (adreed-msft) [inline e2etest/newe2e_task_resourcemanagement.go:224]: Q: Is this function designed to

- Check all errors in the list are present
or
- Check at least one error in the list is present

**Review by adreed-msft — APPROVED** [review]
Looks good to me, questions about testing, though.

**Comment by adreed-msft** [comment]
Another thought-- Does this catch things like a job failing after it begins running?

**Review by pranavmalik-msft — COMMENTED** [review]
Have we added tests for list, remove as we discussed ?

File: e2etest/newe2e_task_resourcemanagement.go (hunk 1 above)
Comment (siminsavani-msft) [reply e2etest/newe2e_task_resourcemanagement.go:224]: This is designed to check that at least one error is present. I can rename it to ValidateContainsError or something of the sort.

**Comment by siminsavani-msft** [comment]
> Have we added tests for list, remove as we discussed ?

Was working on other PRs, just added now!

**Comment by siminsavani-msft** [comment]
> Another thought-- Does this catch things like a job failing after it begins running?

Mostly catches the service side errors and spits out related shorthand error code URLs.

**Review by pranavmalik-msft — APPROVED** [review]
LGTM

**Pull request merged**